### PR TITLE
Added .editorconfig

### LINKED ISSUE #2103: Add .editorconfig file
Our style follows the example given on http://editorconfig.org/ for Python and YAML
label: difficulty/low  initiatives/gci
Opened by @mohitkyadav at [Gitter](https://gitter.im/coala/coala?at=59f862f94ff065ac1899d4d1)

### PR DESCRIPTION
**Pull request by Subhash-Saurabh**
Added .editorconfig file
Closes #2103

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
```
Comment (newbazz) [inline .editorconfig]: add a `charset = utf-8` here ;P

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+end_of_line = lf
+insert_final_newline = true
```
Comment (newbazz) [inline .editorconfig]: you can add a `max_line_length` feature as well.

Comment (Subhash-Saurabh) [reply .editorconfig]: @newbazz  What should be the value of `max_line_length`.

File: .editorconfig (hunk 1 above)
Comment (Subhash-Saurabh) [reply .editorconfig]: It is already present on line 12 .

Comment (newbazz) [reply .editorconfig]: `[*]` means the file present in the present depth, so you need to add this.

File: .editorconfig (hunk 2 above)
Comment (newbazz) [reply .editorconfig]: same goes with this.

Comment (newbazz) [reply .editorconfig]: have a read on what this file does ;)

**Comment by Makman2** [comment]
Closing due to inactivity.

**Pull request closed without merging**